Add React base 404 page

CodeWritingCow opened the pull request:
Converted 404 page from Handlebars to React component,`<NotFound/>`

Modified `dist/index.html` to load bundle.js and style.css from absolute path rather than relative path. This prevents the client from looking for those files in the wrong directory if users visit a nested route. Previously, if a user visited `localhost:8080/pagenotfound/pagenotfound`, React tried to load `localhost:8080/pagenotfound/bundle.js` instead of the correct path, `localhost:8080/bundle.js`.

CodeWritingCow committed:
React 404 Page

- created <NotFound/>
- imported React-Parallax and included image
- added wildcard React Router path in Index.jsx for catching
requests for non-existent pages
- removed .hbs res.render in Express error-controller.js

CodeWritingCow committed:
Merge branch 'react-base' of https://github.com/CodeWritingCow/NYC-restaurant-inspection-records into react-base-404-page

- Update branch with new commits from react-base main branch

CodeWritingCow committed:
Merge branch 'react-base' of https://github.com/CodeWritingCow/NYC-restaurant-inspection-records into react-base-404-page

Update branch with new commits from react-base main branch

CodeWritingCow committed:
Merge branch 'react-base' of https://github.com/CodeWritingCow/NYC-restaurant-inspection-records into react-base-404-page

Update branch with new commits from react-base main branch

CodeWritingCow committed:
Load bundle.js from absolute path

Modified index.html to load bundle.js from absolute path instead
of relative path. This prevents the client from looking for bundle.js
in the wrong directory if users visit a nested route.

Previously, if a user visited localhost:8080/pagenotfound/pagenotfound,
React tried to load localhost:8080/pagenotfound/bundle.js instead of
the correct path, localhost:8080/bundle.js.

badwolfbit commented:
@CodeWritingCow there is only a stylistic issue that needs to be addressed before merging

### Original: 

<img width="1440" alt="Screenshot 2019-07-16 14 30 53" src="https://user-images.githubusercontent.com/7663419/61320306-2742b080-a7d7-11e9-9f9d-45d05ca78e04.png">

### PR #58:

<img width="1440" alt="Screenshot 2019-07-16 14 30 47" src="https://user-images.githubusercontent.com/7663419/61320313-2b6ece00-a7d7-11e9-81c3-4eab34ff3ad0.png">

CodeWritingCow committed:
Modify margin top of parallax content

Add margin to move content closer to parallax image's vertical center.

Also, remove class "hide-on-small-only" from Parallax component, so
Parallax appears on small screens.

CodeWritingCow commented:
@omenwolf I addressed the styling issue by increasing the margin top of Parallax's content. The React 404 page looks identical to the original Handlebars page.

badwolfbit commented:
@CodeWritingCow Looks great, will merge
